File: fib/fib.c
#include <stdio.h>
#include <stdlib.h>

unsigned int fib(unsigned int n)
{
    if (n < 2) return n;

    unsigned int* table = malloc((n + 1) * sizeof(unsigned int));
    table[0] = 0;
    table[1] = 1;
    unsigned int i; for (i = 2; i <= n; i++) table[i] = table[i-2] + table[i - 1];
    unsigned int res = table[n];
    free(table);
    return res;
}

int main(int argc, char const *argv[])
{
	int res = fib(atoi(argv[1]));
	printf("%d\n", res);
	return 0;
}

File: fib/Fib.java
public class Fib {
	public static Integer fib(int n) {
		if (n < 2)
			return n;
		Integer[] table = new Integer[n + 1];
		table[0] = 0;
		table[1] = 1;
		for (int i = 2; i <= n; i++) {
			table[i] = table[i - 1] + table[i - 2];
		}
		return table[n];
	}

	public static void main(String[] args) {
		Integer res = Fib.fib(Integer.parseInt(args[0]));
		System.out.println(res);
	}
}
